src/ts/InternalBitmovinYospacePlayer.ts

```diff
@@ -1503,13 +1510,16 @@ export class InternalBitmovinYospacePlayer implements BitmovinYospacePlayerAPI {
   };
 
   private calculateAdParts(): void {
-    const adParts = this.session?.getAdBreaksByType(BreakType.LINEAR).map((adBreak) => ({
-      start: toSeconds(adBreak.getStart()),
-      end: toSeconds(adBreak.getStart()) + toSeconds(adBreak.getDuration()),
-      adBreak: adBreak,
-    }));
-    if (adParts) {
-      this.adParts = adParts;
+    const linearAdBreaks = this.session?.getAdBreaksByType(BreakType.LINEAR);
+    if (linearAdBreaks !== null) {
```

Comment:
Again, `linearAdBreaks` could be `undefined` (if `this.session` is null/undefined). Let's check for undefined and null:

```suggestion
    if (linearAdBreaks) {
```

Reply:
thank you; updated w/ suggested changes..